### DISCUSSION TITLE
How does the whole keybinding context system work?

### DISCUSSION THREAD

**Discussion by test3211234** [zed-industries/zed#15401] [Feature Requests]
I've been looking through the codebase for an hour and still have no idea how it works.
- How does the editor decide whether things like "Editor" and "VimControl" and "PromptLibrary" and "Pane" and "Markdown" are true or false and how does it change that?
- Can multiple be true at once?
- How do they differ from snake_case variables like "mode" and "not_editing", in how they're set and changed? How are they set and changed? Why is there a difference in naming and what is the difference even between the uppercase and lowercase variables?
- This isn't a code question, but what does the > operator do?

 I noticed there's a .key_context function on Divs, but looking into that just made my head hurt even more. Thanks.

**Comment by neunato** [comment]
> How does the editor decide whether things like "Editor" and "VimControl" and "PromptLibrary" and "Pane" and "Markdown" are true or false and how does it change that?

it tells you which element has focus

> Can multiple be true at once?

yes because elements can be nested (both parent and child have focus)

> How do they differ from snake_case variables like "mode" and "not_editing", in how they're set and changed? How are they set and changed? Why is there a difference in naming and what is the difference even between the uppercase and lowercase variables?

those inform you of additional state in the app

> This isn't a code question, but what does the > operator do?

checks for an immediate child of an element

**Reply by test3211234** [reply]
Thanks. Do you have any idea how the editor internally knows what has focus?

**Reply by neunato** [reply]
I know nothing about the internals and can't see why that would be relevant to you

anyway as said in https://github.com/zed-industries/zed/issues/14718#issuecomment-2254099099 - I'm just making assumptions and would also appreciate clarification from the developers

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
